src/engraving/io/mscreader.cpp

```diff
@@ -0,0 +1,379 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "mscreader.h"
+
+#include <QXmlStreamReader>
+#include <QFile>
+#include <QFileInfo>
+#include <QDir>
+#include <QDirIterator>
+
+#include "thirdparty/qzip/qzipreader_p.h"
+
+#include "log.h"
+
+//! NOTE The current implementation resolves files by extension.
+//! This will probably be changed in the future.
+
+using namespace mu::engraving;
+
+MscReader::MscReader(const Params& params)
+    : m_params(params)
+{
+}
+
+MscReader::~MscReader()
+{
+    close();
+}
+
+void MscReader::setParams(const Params& params)

```

Comment:
Maybe it's better to not allow to change parameters of an existing MscReader instance and just leave an explicit constructor MscReader(const Params& params)?

Reply:
Maybe, but sometimes it may not be convenient. For example, if we want to make the reader a member of the class and use one instance, setting parameters to it.